unity/Assets/Scripts/CanOpen_Object.cs

```diff
@@ -97,529 +78,218 @@ void Start ()
-                        //only aply openPercentage modifier to components of vector3 that actually change
-                        if(openPositions[i].x - closedPositions[i].x != Mathf.Epsilon)
-                        {
-                            lerpToPosition.x = ((openPositions[i].x - closedPositions[i].x) * openPercentage) + closedPositions[i].x;
-                        }
-
-                        if(openPositions[i].y - closedPositions[i].y != Mathf.Epsilon)
-                        {
-                            lerpToPosition.y = ((openPositions[i].y - closedPositions[i].y) * openPercentage) + closedPositions[i].y;
-                        }
-
-                        if(openPositions[i].z - closedPositions[i].z != Mathf.Epsilon)
-                        {
-                            lerpToPosition.z = ((openPositions[i].z - closedPositions[i].z) * openPercentage) + closedPositions[i].z;
-                        }
-
-                        iTween.MoveTo(MovingParts[i], iTween.Hash(
-                        "position", lerpToPosition,
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear"));
-                    }
-
-				}
+        startOpenness = currentOpenPercentage;
+        for (int i = 0; i < MovingParts.Length; i++) {
+            Hashtable args = new Hashtable() {
+                {"islocal", true},
+                {"time", animationTime},
+                {"easetype", "linear"}
+            };
+
+            // we are on the last moving part here
+            if (i == MovingParts.Length - 1) {
+                args["onCompleteTarget"] = gameObject;
+                args["oncompleteparams"] = isOpen ? 0f : openPercentage;
+            }
 
-                //scale with Y axis
-                else if(movementType == MovementType.ScaleY)
-                {
-                    //we are on the last loop here
-                    if(i == MovingParts.Length -1)
-                    {
-                        iTween.ScaleTo(MovingParts[i], iTween.Hash(
-                        "scale", new Vector3(openPositions[i].x, closedPositions[i].y + (openPositions[i].y - closedPositions[i].y) * openPercentage, openPositions[i].z),
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear", "onComplete", "setisOpen", "onCompleteTarget", gameObject,
-                        "oncompleteparams", openPercentage));
-                    }
-
-                    else
-                    iTween.ScaleTo(MovingParts[i], iTween.Hash(
-                    "scale", new Vector3(openPositions[i].x, closedPositions[i].y + (openPositions[i].y - closedPositions[i].y) * openPercentage, openPositions[i].z),
-                    "islocal", true,
-                    "time", animationTime,
-                    "easetype", "linear"));
+            // let's open the object!
```

Comment:
Note: all of these animations are _exactly_ the same as how they started. They are just refactored with `args` so they aren't as repetitive.

Reply:
looks good to me